import/sql/get_station_importance.sql

```diff
@@ -71,82 +62,117 @@ CREATE OR REPLACE VIEW station_nodes_stop_positions_rel_count AS
+  ) AS facilities
+  GROUP BY cluster_id, name, station, railway_ref, uic_ref, railway;
+
+CREATE INDEX IF NOT EXISTS stations_clustered_station_ids
+  ON stations_clustered
+    USING gin(station_ids);
+
+CREATE OR REPLACE VIEW stations_with_route_count AS
+  SELECT
+    id,
+    max(route_count) as route_count
+  FROM (
+    SELECT
+      id,
+      COUNT(DISTINCT route_id) AS route_count
     FROM (
       SELECT
-        *,
-        ST_ClusterDBSCAN(way, 400, 1) OVER (PARTITION BY name, station, railway_ref, uic_ref, railway) AS cluster_id
-      FROM (
-        SELECT MIN(id) as id, MIN(osm_id) as osm_id, name, station, railway_ref, uic_ref, railway, ARRAY_LENGTH(ARRAY_AGG(DISTINCT route_id), 1) AS route_count, name_tags, way
-        FROM (
-          SELECT id, osm_id, name, station, railway_ref, uic_ref, railway, UNNEST(route_ids) AS route_id, name_tags, way
-          FROM station_nodes_stop_positions_rel_count
-
-          UNION ALL
-
-          SELECT id, osm_id, name, station, railway_ref, uic_ref, railway, UNNEST(route_ids) AS route_id, name_tags, way
-          FROM station_nodes_platforms_rel_count
-        ) AS a
-        GROUP BY name, station, railway_ref, uic_ref, railway, way, name_tags
-
-        UNION ALL
-
-        SELECT id, osm_id, name, station, railway_ref, uic_ref, railway, 0 AS route_count, name_tags, way
-        FROM stations
-        WHERE railway IN ('station', 'halt', 'tram_stop', 'service_station', 'yard', 'junction', 'spur_junction', 'crossover', 'site')
-     ) AS grouped_facilities
-    ) AS facilities
-    GROUP BY name, station, railway_ref, uic_ref, railway, cluster_id
-    -- ORDER BY is required to ensure that the larger route_count is used.
-    ORDER BY name, station, railway_ref, uic_ref, railway, route_count DESC
-  ) as source_facilities;
-
-CREATE INDEX IF NOT EXISTS stations_with_route_counts_center_index
-  ON stations_with_route_counts
-  USING GIST(center);
-
-CREATE INDEX IF NOT EXISTS stations_with_route_counts_buffered_index
-  ON stations_with_route_counts
-  USING GIST(buffered);
+        id,
+        UNNEST(route_ids) AS route_id
+      FROM station_nodes_stop_positions_rel_count
+
+      UNION ALL
+
+      SELECT
+        id,
+        UNNEST(route_ids) AS route_id
+      FROM station_nodes_platforms_rel_count
+    ) stations_that_have_routes
+    GROUP BY id
+
+    UNION ALL
+
+    SELECT
+      id,
+      0 AS route_count
+    FROM stations
+    WHERE
+      railway IN ('station', 'halt', 'tram_stop', 'service_station', 'yard', 'junction', 'spur_junction', 'crossover', 'site')
+  ) all_stations_with_route_count
+  GROUP BY id;
+
+-- Final table with station nodes and the number of route relations
+-- needs about 3 to 4 minutes for whole Germany
+-- or about 20 to 30 minutes for the whole planet
+CREATE MATERIALIZED VIEW IF NOT EXISTS grouped_stations_with_route_count AS
```

Comment:
Major change in database structure for stations: the clustering and the route count now have different views that are combined in the final `grouped_stations_with_route_count` view.